site/app/controllers/HomePageController.php

```diff
@@ -41,6 +42,53 @@ public function showLibrary() {
     }
 
     /**
+     * @Route("/homework/library/search", methods={"GET"})
+     * @return Response
+     */
+    public function searchLibrary($path = "/usr/local/submitty/library") {
+        $gradeable_ids = FileUtils::getAllDirs($path);
+        return Response::JsonOnlyResponse(
+            JsonResponse::getSuccessResponse($gradeable_ids)
+        );
+	}
+
+    /**
+     * @Route("/homework/library/search/{query}", methods={"GET"})
+     * @return Response
+     */
+    public function searchLibraryWithQuery($query, $path = "/usr/local/submitty/library") {
+        $gradeable_ids = FileUtils::getDirWithText($path, $query);
+        return Response::JsonOnlyResponse(
+            JsonResponse::getSuccessResponse($gradeable_ids)
+        );
+	}
+
+    /**
+     * @Route("/homework/library/search_gradeable/{query}", methods={"GET"})
+     * @return Response
+     */
+    public function searchLibraryGradeableWithQuery($query, $path = "/usr/local/submitty/library") {
+        $config = FileUtils::getDetails($path, $query);
+        return Response::JsonOnlyResponse(
+            JsonResponse::getSuccessResponse($config)
+        );
+	}
+    
+    /**
+     * Display the LibraryManagerView to the instructor/admin.
+     *
+     * @Route("/homework/library/manage", methods={"GET"})
+     * @return Response
+     */
+    public function showLibraryManager() {
+        $this->core->getOutput()->addBreadcrumb("Manage");
+        $this->core->getOutput()->addInternalCss('admin-gradeable.css');
```

Comment:
I think the above two lines should go into the view.

Reply:
yeah this got moved like a month ago. The library kickoff branch needs to be merged into this again